src/main/resources/schema.sql

```diff
@@ -1,57 +1,78 @@
+CREATE TABLE IF NOT EXISTS friendship
+(
+    user_id        INTEGER NOT NULL REFERENCES users (id) ON DELETE CASCADE ON UPDATE CASCADE,
+    friend_user_id INTEGER NOT NULL REFERENCES users (id) ON DELETE CASCADE ON UPDATE CASCADE,
+    PRIMARY KEY (user_id, friend_user_id)
 );
 
 -- Таблица с рейтингами фильмов
-CREATE TABLE IF NOT EXISTS mpa (
-        mpa_id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
-        name VARCHAR(50) NOT NULL
+CREATE TABLE IF NOT EXISTS mpa
+(
+    mpa_id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+    name   VARCHAR(50) NOT NULL
 );
 
 -- Таблица с фильмами
-CREATE TABLE IF NOT EXISTS films (
-        id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
-        name VARCHAR(50) NOT NULL CHECK (name <> ''),
-        description VARCHAR(255) NOT NULL,
-        duration INTEGER CHECK (duration > 0),
-        releaseDate DATE,
-        mpa_id INTEGER NOT NULL,
-        FOREIGN  KEY (mpa_id) REFERENCES mpa(mpa_id),
-        CONSTRAINT exist_film_name UNIQUE (name)
+CREATE TABLE IF NOT EXISTS films
+(
+    id          INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+    name        VARCHAR(50)  NOT NULL CHECK (name <> ''),
+    description VARCHAR(255) NOT NULL,
+    duration    INTEGER CHECK (duration > 0),
+    releaseDate DATE,
+    mpa_id      INTEGER      NOT NULL,
+    FOREIGN KEY (mpa_id) REFERENCES mpa (mpa_id),
+    CONSTRAINT exist_film_name UNIQUE (name)
 );
 
 -- Таблица со всеми жанрами
-CREATE TABLE IF NOT EXISTS genres (
-        genre_id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
-        name VARCHAR(50)
+CREATE TABLE IF NOT EXISTS genres
+(
+    genre_id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+    name     VARCHAR(50)
 );
 
 -- Таблица с жанрами фильмов
-CREATE TABLE IF NOT EXISTS film_genres (
-        film_id INTEGER REFERENCES films(id) ON delete CASCADE,
-        genre_id INTEGER REFERENCES genres(genre_id) ON delete CASCADE,
-        PRIMARY KEY(film_id, genre_id)
+CREATE TABLE IF NOT EXISTS film_genres
+(
+    film_id  INTEGER REFERENCES films (id) ON delete CASCADE,
+    genre_id INTEGER REFERENCES genres (genre_id) ON delete CASCADE,
+    PRIMARY KEY (film_id, genre_id)
 );
 
 -- Таблица с лайками пользователей
-CREATE TABLE IF NOT EXISTS film_likes (
-        film_id INTEGER REFERENCES films(id) ON delete CASCADE,
-        user_id INTEGER REFERENCES users(id) ON delete CASCADE,
-        PRIMARY KEY(film_id, user_id)
+CREATE TABLE IF NOT EXISTS film_likes
+(
+    film_id INTEGER REFERENCES films (id) ON delete CASCADE,
+    user_id INTEGER REFERENCES users (id) ON delete CASCADE,
+    PRIMARY KEY (film_id, user_id)
+);
+
+-- Таблица с режиссерами
+CREATE TABLE IF NOT EXISTS directors
+(
+    director_id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+    name        VARCHAR(50) NOT NULL
+);
+
+-- Таблица с фильмами режиссеров
+CREATE TABLE IF NOT EXISTS film_directors
+(
+    director_id INTEGER,
+    film_id     INTEGER
```

Comment:
Я бы добавил каскадное удаление:
```suggestion
CREATE TABLE IF NOT EXISTS film_directors
(
    director_id INTEGER REFERENCES directors (director_id) ON delete CASCADE,
    film_id INTEGER REFERENCES films (id) ON delete CASCADE
```

Reply:
Согласен.